tests/test_cnes_sonarqube.py

```diff
@@ -0,0 +1,238 @@
+
+Pytest documentation: https://docs.pytest.org/en/stable/contents.html
+"""
+
+import os
+import platform
+import re
+import subprocess
+import time
+
+import docker
+import requests
+
+
+class TestCNESSonarQube:
+    """This class test the lequal/sonarqube image.
+    It does not build it.
+    Tests can be parametered with environment variables.
+
+    Environment variables:
+        RUN: whether or not to run a container, default "yes", if you
+             already have a running container, set it to "no" and provide
+             information through the other variables.
+        SONARQUBE_CONTAINER_NAME: the name to give to the container running
+                                  the image.
+        SONARQUBE_ADMIN_PASSWORD: the password of the admin account.
+        SONARQUBE_URL: URL of lequal/sonarqube container if already running
+                       without trailing / from the host.
+                       e.g. http://localhost:9000
+
+    The docstring of each test is the user story it tests.
+    """
+    # Class variables
+    RUN = os.environ.get('RUN', "yes")
+    SONARQUBE_CONTAINER_NAME = os.environ.get('SONARQUBE_CONTAINER_NAME', "lequalsonarqube")
+    SONARQUBE_ADMIN_PASSWORD = os.environ.get('SONARQUBE_ADMIN_PASSWORD', "adminpassword")
+    SONARQUBE_URL = os.environ.get('SONARQUBE_URL', "http://localhost:9000")
+
+    # Functions
+    @classmethod
+    def wait_cnes_sonarqube_ready(cls, container_name: str, tail = "all"):
+        """This function waits for SonarQube to be configured by
+        the configure.bash script.
+
+        :param container_name: name of the container running lequal/sonarqube
+        :param tail: forwarded to docker logs
+        """
+        docker_client = docker.from_env()
+        while b'[INFO] CNES SonarQube: ready!' not in docker_client.containers.get(container_name).logs(tail=tail):
+            time.sleep(10)
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the tests
+        Launch a container and wait for it to be up
+        """
+        docker_client = docker.from_env()
+        # Launch a CNES SonarQube container
+        if cls.RUN == "yes":
+            print(f"Launching lequal/sonarqube container (name={cls.SONARQUBE_CONTAINER_NAME})...")
+            docker_client.containers.run("lequal/sonarqube:latest",
+                name=cls.SONARQUBE_CONTAINER_NAME,
+                detach=True,
+                auto_remove=True,
+                environment={"SONARQUBE_ADMIN_PASSWORD": cls.SONARQUBE_ADMIN_PASSWORD},
+                ports={9000: 9000})
+        else:
+            print(f"Using container {cls.SONARQUBE_CONTAINER_NAME}")
+        # Wait for the SonarQube server inside it to be set up
+        print(f"Waiting for {cls.SONARQUBE_CONTAINER_NAME} to be up...")
+        cls.wait_cnes_sonarqube_ready(cls.SONARQUBE_CONTAINER_NAME)
+
+    @classmethod
+    def teardown_class(cls):
+        """Stop the container"""
```

Comment:
Codacy found an issue: [First line should end with a period, question mark, or exclamation point (not 'r') (D415)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)